system-int-test/test/cmr/system_int_test/search/subscription/concept_retrieval_test.clj

```diff
@@ -0,0 +1,124 @@
+(ns cmr.system-int-test.search.subscription.concept-retrieval-test
+  "Integration test for service retrieval via the following endpoints:
+
+  * /concepts/:concept-id
+  * /concepts/:concept-id/:revision-id"
+  (:require
+   [cheshire.core :as json]
+   [clojure.test :refer :all]
+   [cmr.common.mime-types :as mt]
+   [cmr.mock-echo.client.echo-util :as e]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.utils.subscription-util :as subscription]))
+
+(use-fixtures
+ :each
+ (join-fixtures
+  [(ingest/reset-fixture {"provguid1" "PROV1"})
+   subscription/grant-all-subscription-fixture]))
+
+(defn- assert-retrieved-concept
+  "Verify the retrieved concept by checking against the expected subscription name,
+  which we have deliberately set to be different for each concept revision."
+  [concept-id revision-id accept-format expected-subscription-name]
+  (let [{:keys [status body]} (search/retrieve-concept
+                               concept-id revision-id {:accept accept-format})]
+    (is (= 200 status))
+    (is (= expected-subscription-name
+           (:Name (json/parse-string body true))))))
+
+(defmulti handle-retrieve-concept-error
+  "Execute the retrieve concept call with the given parameters and returns the status and errors
+  based on the result format."
+  (fn [concept-id revision-id accept-format]
+    accept-format))
+
+(defmethod handle-retrieve-concept-error mt/umm-json
+  [concept-id revision-id accept-format]
+  (let [{:keys [status body]} (search/retrieve-concept concept-id revision-id
+                                                       {:accept accept-format})
+        errors (:errors (json/parse-string body true))]
+    {:status status :errors errors}))
+
+(defmethod handle-retrieve-concept-error :default
+  [concept-id revision-id accept-format]
+  (search/get-search-failure-xml-data
+   (search/retrieve-concept concept-id revision-id {:accept accept-format
+                                                    :throw-exceptions true})))
+
+(defn- assert-retrieved-concept-error
+  "Verify the expected error code and error message are returned when retrieving the given concept."
+  [concept-id revision-id accept-format err-code err-message]
+  (let [{:keys [status errors]} (handle-retrieve-concept-error
+                                 concept-id revision-id accept-format)]
+    (is (= err-code status))
+    (is (= [err-message] errors))))
```

Comment:
Rather than copy paste these code from `cmr.system-int-test.search.service.concept-retrieval-test`, can you refactor these into an existing helper namespace and just call them from there?

Reply:
Maybe a separate ticket to refactor all the service and variable and subscription ones.